superset/db_engine_specs/hive.py

```diff
@@ -343,7 +346,7 @@ def select_star(
         latest_partition: bool = True,
         cols: Optional[List[Dict[str, Any]]] = None,
     ) -> str:
-        return BaseEngineSpec.select_star(
+        return super(PrestoEngineSpec, cls).select_star(
```

Comment:
As `HiveEngineSpec` already extends `PrestoEngineSpec`, this wouldn't really be needed. It is my understanding, that the reason for calling `BaseEngineSpec` here is because we specifically don't want to use the default behaviour of `PrestoEngineSpect`, but rather what's defined in `BaseEngineSpec`. Not being familiar with how Hive/Presto works in this case someone with deeper knowledge of their respective intricacies would probably need to check this.

Reply:
So, you're right that we want to call the `BaseEngineSpec` version of `select_star`, but that function calls `where_latest_partition` which we need from the `HiveEngineSpec`. the Presto version of `select_star` doesn't do much more than call `super().select_star`, but I think updating it to `super(BaseEngineSpec, cls)` is probably the right move.

Reply:
Actually, this was right before. using `super(PrestoEngineSpec, cls)` calls `HiveEngineSpec`'s grandparent's (`BaseEngineSpec`) version of `select_star`. Then, by passing in `cls` afterwards, we go back to running Hive's version of functions after the base class's `select_star` gets called